### ISSUE TITLE
TR2X bug: Some music triggers cause game to freeze briefly

### ISSUE THREAD

**Issue by aredfan** [LostArtefacts/TRX#1597]
**Dev snapshot:** 0.1.1-15-g19a94d0

_____________________________

When Lara crosses the music triggers marked in the screenshots below, the game freezes very briefly.

This issue is not present in 0.1.1

[saves.zip](https://github.com/LostArtefacts/TR2X/files/15186532/saves.zip)

![1](https://github.com/LostArtefacts/TRX/assets/94776985/30e375fd-bc4a-40cc-9fd3-7fe541aeb61c)

![2](https://github.com/LostArtefacts/TRX/assets/94776985/87df58d7-2b80-4fd7-acbc-6222abc5033f)

**Comment by rr-** [comment]
Shit, that's what we get for using single threaded audio decoding.
@aredfan what happens if you remove either `audio/cdaudio.*`, or `music` files? (Renaming the directories alone will suffice)

**Comment by aredfan** [comment]
@rr- 

- If I only use the music folder with all the separate soundtracks, there is no lag.
- If I only use the audio folder (cdaudio.dat and cdaudio.mp3), the lag happens.

**Comment by rr-** [comment]
Thanks. The results are a bit confusing to me. The game defaults to using `music/` folder, which in the last message is said to be lag-free. Can you confirm if there was still lag present when you had both types of files available (in which case it should prioritize the `music/` directory)?

**Comment by aredfan** [comment]
@rr- Yep, the music folder takes priority on my setup too when both folders are present.  There's no lag with this arrangement.

Actually, the reason I ran into this issue is because I used a TR2 disc re-release which doesn't come with a music folder, so I used Multi Patch which provides a audio folder instead.

I installed my Steam copy today to test your scenarios because it had both music and audio folders.  I used the disc version before today because it was already installed on my PC.  I can understand the confusion caused so I apologise for that.

**Comment by rr-** [comment]
No worries – thanks for clearing things up. We'll have to decode in a background thread but it's a significant effort.

**Comment by aredfan** [comment]
Closing this as it looks like TR2X's main focus is supporting the music folder when taking into account the gong fix (#1725).  We can reopen this issue in the future if needed.
